### PR TITLE
MockProver test utililities

### PR DESCRIPTION
**Pull request by naure**
This adds support for new unittests in zkevm-circuits, which need two things:
* To read what the circuit under test assigned.
* To escape the abstract Value type to test concrete values.

### CODE REVIEW

**Comment by naure** [comment]
> Although there aren't too many cases that could benefit from advice_values

Or it is the other way around! There aren’t a lot of regression tests today because there is no easy way to write them without this sort of API.

File: halo2_proofs/src/circuit/value.rs [hunk 1]
```diff
@@ -696,3 +696,11 @@ impl<F: Field> Value<Assigned<F>> {
...
     }
 }
+
+/// Utilities for tests and dev tools.
+pub mod value_dev {
+    /// Get the contained known value, or panic. For tests only.
+    pub fn unwrap_value<T>(v: super::Value<T>) -> T {
+        v.inner.unwrap()
+    }
+}
```
[the remark this answers is not in the document]
Comment (naure) [reply halo2_proofs/src/circuit/value.rs]: Removed.

**Review by han0110 — APPROVED** [review]
LGTM!

**Review by ed255 — APPROVED** [review]
LGTM!

**Pull request merged**